pkg/lib/fieldgroups/database/database_validator.go

```diff
@@ -28,8 +31,38 @@ func (fg *DatabaseFieldGroup) Validate(opts shared.Options) []shared.ValidationE
 		ca = fg.DbConnectionArgs.Ssl.Ca
 	}
 
+	sslrootcertTmpPath := fg.DbConnectionArgs.SslRootCert
+	if fg.DbConnectionArgs.SslMode == "verify-full" || fg.DbConnectionArgs.SslMode == "verify-ca" {
+		// Write database.pem needed for db validation, if any, to a temp file
+		tmpCert, err := ioutil.TempFile("/tmp", "database.*.pem")
+		if err != nil {
+			newError := shared.ValidationError{
+				Tags:       []string{"DB_URI"},
+				FieldGroup: fgName,
+				Message:    "Could not connect to database. Error: " + err.Error(),
```

Comment:
This error message is misguiding. We haven't attempted to connect to the database.

Reply:
Updated